docs/design.md

```diff
@@ -38,13 +58,168 @@ spec:
+```
+
+The example Model's backing storage would end up being:
+
+```sh
+gs://abc123-substratus-artifacts/f94a0d128bcbd9c1b824e9e5572baf86/model/
+gs://abc123-substratus-artifacts/f94a0d128bcbd9c1b824e9e5572baf86/logs/
+```
+
+The Model would report this URL in its status:
+
+```yaml
+kind: Model
+# ...
+status:
+  url: gs://abc123-substratus-artifacts/f94a0d128bcbd9c1b824e9e5572baf86
+```
+
+#### Reconcile Logic
+
+Pseudo-reconciler logic for a Model:
+
+```
+if .status.url != "" {
+  return
+}
+
+url := "{bucket}/{hash}"
+if sci.bucketObjectExists(url + "/completed.json") {
+  # Use the bucket as the source of truth if .status.url did not exist.
+  # If "completed.json" exists, it means that a given long-running
+  # training/loading Job completed successfully.
+  updateStatus(url)
+  return
+}
+
+# ... Looks like the model should be imported/trained/etc.
+
+runJob()
+```
+
+#### Mount Points
+
+All mount points will are made under a standardized `/content` directory which should correspond to the `WORKDIR` of a Dockerfile. This works well for Jupyter notebooks which can be run with `/content` set as the serving directory: all relevant mounts will be populated on the file explorer sidebar.
+
+The `logs/` directories below are used to store the notebook cell output, python logs, tensorboard stats, etc. These directories are mounted as read-only in Notebooks to explore the output of other background jobs.
+
+##### Dataset (importing)
+
+```
+/content/params.json  # Populated from .spec.params (also available as env vars).
+
+/content/data/        # Mounted RW: initially empty dir to write new files
+/content/logs/        # Mounted RW: initially empty dir to write new files
+```
+
+##### Model (importing)
+
+```
+/content/params.json  # Populated from .spec.params (also available as env vars).
+
+/content/model/       # Mounted RW: initially empty dir to write new files
+/content/logs/        # Mounted RW: initially empty dir to write new files
+```
+
+##### Model (training)
+
+```
+/content/params.json        # Populated from .spec.params (also available as env vars).
+
+/content/data/              # Mounted RO: from .spec.trainingDataset
+
+/content/saved-model/       # Mounted RO: from .spec.baseModel
+
+/content/model/             # Mounted RW: initially empty dir for writing new files
+/content/logs/              # Mounted RW: initially empty dir for writing logs
+```
+
+##### Notebook
+
+NOTE: The `saved-model/` directory is the same as the container for the Model object when `.baseModel` is specified. This allows for easy development of Model training code.
+
+```
+/content/params.json        # Populated from .spec.params (also available as env vars).
+
+/content/data/              # Mounted RO: from .spec.dataset
+/content/data-logs/         # Mounted RO: from .spec.dataset
+
+/content/saved-model/       # Mounted RO: from .spec.model
+/content/saved-model-logs/  # Mounted RO: from .spec.model
```

Comment:
These are useful to see how the model was trained or how the dataset was originally built yes?

Reply:
Yep, notebooks are for exploration and iteration. This allows users to trace back the origin of the model which is useful IMO.